client/web/src/components/views/device-details-view.tsx

```diff
@@ -210,3 +199,30 @@ export default function DeviceDetailsView({
     </>
   )
 }
+
+function DisconnectDialog() {
+  const api = useAPI()
+  const [, setLocation] = useLocation()
+
+  return (
+    <Dialog
+      className="max-w-md"
+      title="Disconnect"
+      trigger={<Button sizeVariant="small">Disconnect…</Button>}
+    >
+      <Dialog.Form
+        cancelButton
+        submitButton="Disconnect"
+        destructive
+        onSubmit={() => api({ action: "logout" }).then(() => setLocation("/"))}
```

Comment:
There's really not much we can do, but setting the location back to `/` isn't likely to do anything is it, since the client is down at this point?